src/components/activities/searchPatientActivity/SearchPatientActivity.tsx

```diff
@@ -60,6 +67,43 @@ const SearchPatientActivity: FunctionComponent<TProps> = ({
     return has(formik.touched, fieldName) ? get(formik.errors, fieldName) : "";
   };
 
+  const handleScrollToElement = () => {
+    const elem = resultsRef.current;
+    window.scrollTo({
+      top: (elem) ? elem.offsetTop - 200 : 0,
+      behavior: 'smooth',
+    }); 
+  }
+
+  const searchResults = () =>{
+    if(showResults) {
+      if(patientSearchResults && patientSearchResults?.length <= 0){
+        return (
+          <div className="searchPatient__results_none">
+            <h4>We couldn't find a match, please try another search.</h4>
+          </div>
+        );
+      } else {
+        if(showLoader) {
+          return (<h5>Is loading...</h5>);
+        } else {
+          return (
+            <div className="searchPatient__results">
+              <div className="searchPatient__results_count">
+                Results: <strong>{patientSearchResults?.length}</strong>
+              </div>
+              <div className="searchPatient__results_list">
+                {patientSearchResults?.map((patient) => (
+                  <PatientSearchItem patient={patient} />
+                ))}
+              </div>
+            </div>
+          );
+        }
+      }
+    }
+  }
```

Comment:
I think these nested conditionals can grow old badly 🤔. Maybe, a nice way to handle these different returning values would be to use a `switch` along with the _redux state_. Something like a _finite automaton_, you know? Try to take a look at how it's done in **NewPatientActivity** with `state.patients.createPatient.status`.